src/olympia/files/utils.py

```diff
@@ -507,6 +507,9 @@ def check_xpi_info(xpi_info, addon=None):
 
     guid_optional = xpi_info.get('is_webextension', False)
 
+    if addon and not guid and guid_optional:
+        xpi_info['guid'] = guid = addon.guid
```

Comment:
Add a comment maybe ? This is starting to be confusing, I understand it now, I'm not sure I will in 6 months :)